### DISCUSSION TITLE
Custom Cloud Parameter?

### DISCUSSION THREAD

**Discussion by PolarBearGod** [CrowdStrike/psfalcon#106] [Q&A]
Before chasing my tail I wanted to throw out this question. We leverage an API proxy. My understanding with how the API proxy works is that is just injects itself as the parent domain and keeps all the URL paths from the original API call.

Could I add a new line(s) under the `$PSBoundParameters.Cloud` variable switch in the `oath2.ps1` script to account for that proxy?

**Comment by bk-cs** [comment]
Not exactly, because PSFalcon doesn't use the standard `Invoke-WebRequest`/`Invoke-RestMethod` options to make its API calls, so the underlying functions wouldn't know what to do with it.

Instead, I recommend setting the proxy manually or adding it as part of your PowerShell profile:
http://woshub.com/using-powershell-behind-a-proxy/

**Comment by PolarBearGod** [comment]
My apologies if I wasn't more clear. 

There is a difference between a **web** proxy and an **API** proxy. While yes I am behind a web proxy that is already accounted for and configured.

An API proxy acts more like a broker for the key and secret and provisions the session token. This isn't something that can be configured in the Powershell Profile, hence the question.

To know more how this works please see here - https://docs.apigee.com/api-platform/fundamentals/understanding-apis-and-api-proxies

If you are still unsure, I am going to be testing and if it proves to work, I will make a pull request.

**Comment by bk-cs** [comment]
Thank you for the clarification, I was unaware of this option. It sounds like it would be as simple as changing the destination hostname, so I don't see why it wouldn't work to have a custom cloud value. Since the `-Cloud` parameter has static values now, maybe it makes sense to have another parameter dedicated to API Proxy use and pick between that or `-Cloud`?

**Comment by PolarBearGod** [comment]
Worked through a number of files but was ultimately unable to get this working the way we really need it to.   
I have modified the following:

* oauth2.psd1 line 22 - need to add in additional parameter for gateway
* oauth2.ps1  line 26 - Need to create "cloud" parameter for the gateway
* parameters.psd1 lines 34 & 40 - Need to comment out token requirements for client_id and client_secret

Even after all of this PSfalcon was having a hard time with understanding the token granted by the proxy. I am definitely not as gifted to run through all of this, so if you don't mind - lets go ahead and close this unless there is demand elsewhere.

Thanks @bk-cs !!
